test/pkg/environment/common/expectations.go

```diff
@@ -584,25 +584,35 @@ func (env *Environment) ExpectDaemonSetEnvironmentVariableUpdated(obj client.Obj
 	env.ExpectDaemonSetEnvironmentVariableUpdatedWithOffset(1, obj, name, value)
 }
 
-func (env *Environment) ExpectDaemonSetEnvironmentVariableUpdatedWithOffset(offset int, obj client.ObjectKey, name, value string) {
+func (env *Environment) ExpectDaemonSetEnvironmentVariableUpdatedWithOffset(offset int, obj client.ObjectKey, name, value string, containers ...string) {
 	ds := &appsv1.DaemonSet{}
 	ExpectWithOffset(offset+1, env.Client.Get(env.Context, obj, ds)).To(Succeed())
```

Comment:
I know you didn't write this but since you are touching this function, consider changing all the `...WithOffset` calls in favor of the `GinkgoHelper()` at the top of the function (see https://github.com/aws/karpenter/issues/4189)

Reply:
I believe I updated it correctly. Intentionally caused it to fail and the stack trace looked right. I didn't go through and update every offset used in the function (e.g. I didn't refactor `ExpectWithOffset` since it seemed out of scope).